src/javascript/app_2/App/Components/Elements/tabs.jsx

```diff
@@ -0,0 +1,135 @@
+                                </span>
+                            </React.Fragment>
+                        ))
+                    }
+                </TabsWrapper>
+                <TabContents />
+            </div>
+        );
+    }
+};
+
+class TabsWrapper extends React.PureComponent {
+    constructor(props) {
+        super(props);
+        this.state = {
+            sizes: {},
+        };
+        this.els = {};
+        this.getSizes = this.getSizes.bind(this);
+    }
+
+    componentDidMount() {
+        this.getSizes();
+        window.addEventListener('resize', this.getSizes);
+    }
+
+    componentWillUnmount() {
+        window.removeEventListener('resize', this.getSizes);
+    }
+
+    getSizes() {
+        const rootBounds = this.root.getBoundingClientRect();
+        const sizes = {};
+        Object.keys(this.els).forEach((key) => {
+            const el = this.els[key];
+            const bounds = el.getBoundingClientRect();
+
+            const left = bounds.left - rootBounds.left;
+            const right = rootBounds.right - bounds.right;
+
+            sizes[key] = {left, right};
+        });
+        this.setState({sizes});
+    }
+
+    render() {
+        return (
+            <div
+                className='tab-wrapper'
+                ref={el => this.root = el}
+            >
+                {React.Children.map(this.props.children, (child, i) => {
+                    const tab_class = classNames(
+                        'tab',
+                        { 'tab--active': child.key === this.props.active },
+                    );
+                    return (
+                        <div
+                            key={i}
+                            className={tab_class}
+                            onClick={() => {
+                                this.props.onChange(child.key);
+                            }}
+                            ref={el => this.els[child.key] = el}
+                        >
+                            {child}
+                        </div>
+                    );
+                })}
+                <div
+                    className='tab-underline'
+                    style={this.getUnderlineStyle()}
+                />
+            </div>
+        );
+    }
+
+    getUnderlineStyle() {
+        if (this.props.active == null || Object.keys(this.state.sizes).length === 0) {
+            return {left: '0', right: '100%'};
+        }
+        const size = this.state.sizes[this.props.active];
+        return {
+            left      : `${size.left}px`,
+            right     : `${size.right}px`,
+            transition: 'left 0.2s, right 0.25s',
+        };
+    }
+}
+
+TabsWrapper.propTypes = {
+    active      : PropTypes.string,
+    children    : PropTypes.node,
+    toggleDialog: PropTypes.func,
+    onChange    : PropTypes.func,
+};
+
+Tabs.propTypes = {
+    alignment: PropTypes.string,
+    list     : PropTypes.object,
```

Comment:
@zakibinary 
it's better to use `PropTypes.shape` to reveal the properties of the object.